drivers/ata8510/ata8510.c

```diff
@@ -0,0 +1,545 @@
+/*
+ * Copyright (C) 2020 Antonio Galea <antonio.galea@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ata8510
+ * @{
+ *
+ * @file
+ * @brief       Implementation of public functions for ATA8510 radio module
+ *
+ * @author      Antonio Galea <antonio.galea@gmail.com>
+ * @author      Roberto Asquini <bobasquins@gmail.com>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include "xtimer.h"
+#include "thread.h"
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+
+#include "ata8510.h"
+#include "ata8510_params.h"
+#include "ata8510_internals.h"
+
+#define ENABLE_DEBUG (0)
+/*
+ * Warning: to correctly display the debug messages in callbacks,
+ * add CFLAGS+=-DTHREAD_STACKSIZE_IDLE=THREAD_STACKSIZE_DEFAULT to the build
+ * command.
+ */
+#include "debug.h"
+
+#define SPIDEV               (dev->params.spi)
+#define CSPIN                (dev->params.cs_pin)
+#define INTPIN               (dev->params.int_pin)
+#define RESETPIN             (dev->params.reset_pin)
+#define SLEEPPIN             (dev->params.sleep_pin)
+
+static char stack[THREAD_STACKSIZE_DEFAULT];
+static void *_recv_thread(void *arg);
+
+static void _irq_handler(void *arg)
+{
+    ata8510_t *dev = (ata8510_t *)arg;
+
+    switch (dev->internal_state) {
+        case ATA8510_MODE_IDLE:
+            DEBUG("[ata8510] _irq_handler: event received in mode IDLE\n");
```

Comment:
If this is inside an ISR, better use
```suggestion
            DEBUG_PUTS("[ata8510] _irq_handler: event received in mode IDLE");
```